lib/ansible/modules/cloud/docker/docker_container.py

```diff
@@ -1330,6 +1374,77 @@ def _parse_log_config(self):
         except ValueError as exc:
             self.fail('Error parsing logging options - %s' % (exc))
 
+    def _parse_healthcheck(self):
+        '''
+        Return dictionary of healthcheck parameters
+        '''
+        if (not self.healthcheck) or (not self.healthcheck.get('test')):
+            return None, None
+
+        result = dict()
+
+        # all the supported healthecheck parameters
+        options = dict(
+            test='test',
+            interval='interval',
+            timeout='timeout',
+            start_period='start_period',
+            retries='retries'
+        )
+
+        duration_options = ['interval', 'timeout', 'start_period']
+
+        for (key, value) in options.items():
+            if value in self.healthcheck:
+                if value in duration_options:
+                    time = self._convert_duration_to_nanosecond(self.healthcheck.get(value))
+                    if time:
+                        result[key] = time
+                elif self.healthcheck.get(value):
+                    result[key] = self.healthcheck.get(value)
+                    if key == 'test':
```

Comment:
We can reduce the level of indents:

```python
elif option == 'test':
elif option == 'retries':
else:
    result[option] = self.healthcheck.get(option)
```

And I don't thinks someone will be bored enough to specify a `test: ""` to get into the `elif self.healthcheck.get(value):
` check

Reply:
Hmm, that could improve it a bit. It will be still a bit complex, though :)

(Also, never underestimate users which will specify something which doesn't make sense ;) )